### ISSUE TITLE
Why is my BAR1 Memory always occupied?

### ISSUE THREAD

**Issue by RipperPix00** [terminatorul/NvStrapsReBar#89]
I successfully enabled Resizeable Bar a few days ago, but I found that my BAR1 usage is different from everyone else's, my BAR1 is always in a run-out state, even though i'm not running any programs.

My BAR1 usage:
![屏幕截图 2024-06-01 135045](https://github.com/terminatorul/NvStrapsReBar/assets/154543252/d32986a6-d920-4558-8d2c-076dfbbf8361)

Other people's BAR1 usage:
![屏幕截图 2024-06-01 134959](https://github.com/terminatorul/NvStrapsReBar/assets/154543252/4124150b-2279-42eb-b189-220f0664edcc)

**Comment by chknrice** [comment]
have you figured it out? mine does the same

**Comment by Lucide** [comment]
Interesting, I'm observing this too:

* MB: Asrock Z370 Extreme4
* BIOS: L4.21B (CAM support from manufacturer, no additional bios image patching)
* GPU: NVIDIA GeForce RTX 2080 FE
* CPU: Intel Core i7-8086K

```
==============NVSMI LOG==============

Timestamp                                 : Tue Jul 23 12:34:41 2024
Driver Version                            : 560.70
CUDA Version                              : 12.6

Attached GPUs                             : 1
GPU 00000000:01:00.0
    FB Memory Usage
        Total                             : 8192 MiB
        Reserved                          : 203 MiB
        Used                              : 659 MiB
        Free                              : 7331 MiB
    BAR1 Memory Usage
        Total                             : 8192 MiB
        Used                              : 8165 MiB
        Free                              : 27 MiB
    Conf Compute Protected Memory Usage
        Total                             : N/A
        Used                              : N/A
        Free                              : N/A
```

Since I'm unsure of the source of that value, it could have a different meaning than what we are attributing to it. Windows' task manager shows a normal vram usage (as well as GPU-Z).

**Comment by marcmy** [comment]
Same issue here.

**Comment by marcmy** [comment]
Anyone figure this out yet?

**Comment by k1netic88** [comment]
A quick scroll through the more recent posts in working cards, it might be to do with the driver version. 552.55 is the latest driver i could see that has BAR1 usage low.

Im in the same boat, does it affect performance or is it just a difference in reporting?

![image](https://github.com/user-attachments/assets/7e3c4457-e4b0-4d7c-9c6c-aa9efd8359b2)

**Comment by karfelren** [comment]
dunno it's occurred from driver or hardware in my test playing apex does not affect any performance while rebar on still the same 10 - 15% gain performance, my only concern is the microcode from oem (old) it's cause micro stuttering even when not playing game just update the microcode all seamlessly work fine.

**Comment by fidoboy** [comment]
I'm afraid that everyone is having this issue because I'm also observe the same results in nVidia smi tool than the exposed here. Almost all the BAR1 memory is being occupied even when there is no programs running

**Comment by marcmy** [comment]
On a recent driver installation, I noticed that BAR1 wasn't being used. However, even after launching and playing games, the numbers were the same, despite forcing ReBAR through NVIDIA Profile Inspector. A few days later I updated to a newer driver, and BAR1 usage returned to being fully used again.

Perhaps unrelated but I found that enabling ReBAR caused a drop in performance for me, instead of the expected improvement, regardless of whether it was force-enabled via Profile Inspector or not. I haven't been playing many games recently, so my observations are based on Apex Legends and Quake Live. I've resorted to keeping ReBAR disabled for now. Hopefully this can be addressed in the future.

**Comment by AMD-Boii** [comment]
Works well with this version https://www.nvidia.com/download/driverResults.aspx/224483/en-us/

![Снимок экрана 2024-11-06 191718](https://github.com/user-attachments/assets/f000fbd8-9b5c-4fda-916c-76a53011b973)

**Comment by fidoboy** [comment]
> Works well with this version https://www.nvidia.com/download/driverResults.aspx/224483/en-us/
> 
> ![Снимок экрана 2024-11-06 191718](https://private-user-images.githubusercontent.com/89309022/383549435-f000fbd8-9b5c-4fda-916c-76a53011b973.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA5NDk0MjEsIm5iZiI6MTczMDk0OTEyMSwicGF0aCI6Ii84OTMwOTAyMi8zODM1NDk0MzUtZjAwMGZiZDgtOWI1Yy00ZmRhLTkxNmMtNzZhNTMwMTFiOTczLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTA3VDAzMTIwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI3NmEwMjJkMTgwMTg5YjUxMTM4NDJiMDJmNTE0NWNhYzY5ZWZmYzIzMmQ5MjI4MmZjNjAzNDUyZDI4YWNmNTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.8IYx18sKTU-0B78feTJIB7_BY1h7-3P_wP33UM_t3Ok)

You are right. I've just tested it and now the used memory usage is only 2MiB.

**Comment by marcmy** [comment]
That makes sense. I figured out what happened for me was I used to a driver updater and it installed a vulkan beta driver which is based on older drivers (Likely 55x.xx). So that's why it showed "Used: 2 MiB" for me too. But yeah, I didn't see any usage above that when gaming, nor did I see any performance benefit.

**Comment by AMD-Boii** [comment]
So can anybody test it with Dead Space Remake for example? Would it give any performance boost at all?

**Comment by fidoboy** [comment]
It could be interesting to make some tests to see if there is any performance benefits when nvidia-smi.exe displays less memory usage in BAR1

**Comment by fidoboy** [comment]
> Works well with this version https://www.nvidia.com/download/driverResults.aspx/224483/en-us/
> 
> ![Снимок экрана 2024-11-06 191718](https://private-user-images.githubusercontent.com/89309022/383549435-f000fbd8-9b5c-4fda-916c-76a53011b973.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzEwNzQ4MjAsIm5iZiI6MTczMTA3NDUyMCwicGF0aCI6Ii84OTMwOTAyMi8zODM1NDk0MzUtZjAwMGZiZDgtOWI1Yy00ZmRhLTkxNmMtNzZhNTMwMTFiOTczLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTA4VDE0MDIwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJlOTU1YzhiMWNkNDE0ODMxZGVmYmIyZWVkOTkwZjdmMzU5NTI2M2ZjZTI3MTBiMmJmNTdkYWJlM2RiYjliZmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.rDd3h9lUrWcUmYsZK6QUgHbh7kFtkkiiDp2_JxetCls)

It's not sure that these results are equal to "works well". It just means that all that memory is not being used and this is doesn't inmediately means that it have some benefits in performance.

**Comment by PINPAL** [comment]
Any further updates on this? Using a driver that's over a year old doesn't seem ideal?
